### ISSUE TITLE
Logger doesn't seem to work with errors

### ISSUE THREAD

**Issue by jensengar** [dbfannin/ngx-logger#46]
I'm not sure if this is a new thing because I thought it worked before, but it seems whenever I try and log/warn/error an error object it just logs an empty object. My config is the same as you have in the readme, other than a different api url. Here's what happens when I try and log various things in the js console.

```
this.logger.log({msg: 'here we go', custom: 'yup'})
```
gets logged as
```
2018-03-21T00:37:34.131Z [LOG] {
  "msg": "here we go",
  "custom": "yup"
}
```

However if I do
```
error = new Error('Here is my error message');
this.logger.log(error);
```
I get 
```
2018-03-21T00:40:54.334Z [LOG] {}
```

Which I imagine is because that is the result of `JSON.stringify(error)`. Is this a known issue? Are there plans to fix it?

**Comment by th3n3wguy** [comment]
Yeah. This is definitely something that can be handled better. Here is the line of code that is causing your problem: [https://github.com/dbfannin/ngx-logger/blob/master/src/logger.service.ts#L150](https://github.com/dbfannin/ngx-logger/blob/master/src/logger.service.ts#L150)

In my current projects that use this library, I am always passing a `string` as the first parameter and the `Error` or any other objects / arrays / etc. as the rest of the parameters. This also helps me determine where in my code the information is being logged, since the [logging of file / line numbers issue](https://github.com/dbfannin/ngx-logger/issues/21) hasn't been solved yet.

**Comment by th3n3wguy** [comment]
@jensengar => What about the error are you expecting to be logged whenever you log an `Error` object as the first "message" parameter?

**Comment by jensengar** [comment]
Yeah, I think it would be ideal to log the stack because when that error goes up to the server, it would be great to be able to see what *and* where, rather than just what. I think it would probably be sufficient, at least for me if you did something like:
```
if (message instanceof Error) {
 message = message.stack;
} else if (typeof message !== 'string') {
 message = JSON.stringify(message, null, 2);
}
```

Any concerns with that?

**Comment by jensengar** [comment]
I'm happy to PR that if you wish.

**Comment by th3n3wguy** [comment]
@jensengar => The `instanceof` check might not work because of all of the different TYPES of errors that Angular has within its platform, as well as custom errors that you might have created. Here are the class instances we will have to check on (that I am aware of currently). I will update this list as we list more that have to be handled.

- `Error` (general JS type)
- `HttpErrorResponse` - [https://angular.io/api/common/http/HttpErrorResponse](https://angular.io/api/common/http/HttpErrorResponse)

The only other way to get around this is to check for the `.stack` property on the object and assume that it is an instance / extension of the `Error` object.

**Comment by jensengar** [comment]
It's true `.stack` would definitely be a catch-all however if you have an object that *is not and error* but has a stack property, it will not log correctly. As far as custom errors go, if they inherit from the `Error` base class they will still return true with the instance of. See example:

```
class ExError extends Error { }
error = new ExError('here is my ex error'));
error instanceOf Error // returns true
```

For what I need right now though, I can't think of a case where `.stack` would cause any issues if you'd still rather go that route. BTW, thank you for putting this library together, it does everything I could have hoped for.

**Comment by th3n3wguy** [comment]
@jensengar => I'm glad that you like the library, but I am not the one who created it. I am just passionate about maintaining it (for now, that is). @dbfannin is the creator / primary maintainer.

If you want to submit a pull request, I will review it for you. Otherwise, I will get a pull request opened when I get some free time.

**Comment by dbfannin** [comment]
This issue is now fixed with @jensengar's PR. Version ngx-logger@2.0.4. Thanks @th3n3wguy for helping guide this issue.

**Comment by jensengar** [comment]
Has this update been published to npm yet? And if not, when is the planned time for that?

**Comment by dbfannin** [comment]
@jensengar, see comment above. version 2.0.4

**Comment by dbfannin** [comment]
I had an issue with the first time I published to 2.0.3, so I had to unpublish and bump the revision.
